tests/main/snapd-snap/task.yaml

```diff
@@ -166,19 +178,156 @@ execute: |
     EOF
     chmod +x snapcraft-cleanup.sh
     tests.cleanup defer sh -c "$PWD/snapcraft-cleanup.sh"
-
     echo "Ensure we have the fc-cache binaries"
     unsquashfs -ll snapd_spread-test.snap | MATCH bin/fc-cache-v6
     unsquashfs -ll snapd_spread-test.snap | MATCH bin/fc-cache-v7
-
     echo "Ensure we have xdelta3"
     unsquashfs -ll snapd_spread-test.snap | MATCH bin/xdelta3
-
     echo "Ensure we have mksquashfs (and the dependencies)"
     unsquashfs -ll snapd_spread-test.snap | MATCH bin/mksquashfs
     unsquashfs -ll snapd_spread-test.snap | MATCH liblzma.so.5
-
     echo "Ensure we have ld.so.conf"
     unsquashfs -ll snapd_spread-test.snap | MATCH etc/ld.so.conf
     echo "Ensure we have libc"
     unsquashfs -ll snapd_spread-test.snap | MATCH libc.so
+    echo "Ensure we can install the snapd snap"
+    snap install --dangerous snapd_spread-test.snap
+    tests.cleanup defer sh -c "$PWD/snapd-cleanup.sh"
+    echo "Ensure we restarted into the snapd snap"
+    "$TESTSTOOLS"/journal-state match-log  'restarting into "/snap/snapd/'
+    echo "Then we should be able to successfully install a snap"
+    snap install hello-world
+    tests.cleanup defer snap remove --purge hello-world
+    echo "Then hello-world.evil should fail due to snap confinement"
+    if hello-world.evil; then
+        echo "hello-world.evil should fail due to snap confinement"
+        exit 1
+    fi
+    # Replicate the tests from tests/main/interfaces-many-core-provided so
+    # we can exercise that apparmor policy is still effective/correct when using
+    # the snapd snap
+    echo "Given a snap is installed"
+    "$TESTSTOOLS"/snaps-state install-local "$CONSUMER_SNAP"
+    tests.cleanup defer snap remove --purge "$CONSUMER_SNAP"
```

Comment:
Same question.

Reply:
it's not strictly needed but I don't see the harm in having it